[CI] StatelessHollowIndexShardsIT testCloseWhileShardsAreHollowed failing

**Build Scans:**
- [elasticsearch-intake #42631 / part2](https://gradle-enterprise.elastic.co/s/6rez3tijdeslk)
- [elasticsearch-intake #42629 / part2](https://gradle-enterprise.elastic.co/s/2lwl5hm7no6da)
- [elasticsearch-intake #42626 / part2](https://gradle-enterprise.elastic.co/s/sduqogz6g7noo)
- [elasticsearch-intake #42612 / part2](https://gradle-enterprise.elastic.co/s/upeiifv23bx6u)
- [elasticsearch-pull-request #159996 / part-2](https://gradle-enterprise.elastic.co/s/5hniyxuytzyri)
- [elasticsearch-pull-request #159980 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/cm53bj6rgaika)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT.testCloseWhileShardsAreHollowed" -Dtests.seed=3A3F8FE41C801BD -Dtests.locale=ca-AD -Dtests.timezone=Australia/Broken_Hill -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCloseWhileShardsAreHollowed'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: All incoming requests on node [node_t1] should have finished. Expected 0 bytes for requests in-flight but got 250 bytes; pending tasks [[{
  "node" : "node_t1",
  "id" : 13,
  "type" : "persistent",
  "action" : "health-node[c]",
  "status" : {
    "state" : "STARTED"
  },
  "description" : "id=health-node",
  "start_time" : "2026-07-02T17:27:44.668Z",
  "start_time_in_millis" : 1783013264668,
  "running_time" : "1.1m",
  "running_time_in_nanos" : 66977887166,
  "cancellable" : true,
  "cancelled" : false,
  "parent_task_id" : "cluster:4",
  "headers" : { }
},{
  "node" : "node_t1",
  "id" : 86,
  "type" : "transport",
  "action" : "internal:index/shard/recovery/stateless_primary_relocation/start",
  "description" : "",
  "start_time" : "2026-07-02T17:27:45.253Z",
  "start_time_in_millis" : 1783013265253,
  "running_time" : "1.1m",
  "running_time_in_nanos" : 66392306423,
  "cancellable" : false,
  "parent_task_id" : "EUo4xZzkTFe-c3fn4nU0Gg:29",
  "headers" : 
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testCloseWhileShardsAreHollowed
- [main] 2 consecutive failures in step part2
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 6 failures in test testCloseWhileShardsAreHollowed (12.2% fail rate in 49 executions)
- [main] 4 failures in step part2 (23.5% fail rate in 17 executions)
- [main] 4 failures in pipeline elasticsearch-intake (23.5% fail rate in 17 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (7.4% fail rate in 27 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Here's a fuller version of the failure:
```
StatelessHollowIndexShardsIT > testCloseWhileShardsAreHollowed FAILED
    java.lang.AssertionError: All incoming requests on node [node_t1] should have finished. Expected 0 bytes for requests in-flight but got 248 bytes; pending tasks [[{
      "node" : "node_t1",
      "id" : 16,
      "type" : "persistent",
      "action" : "object_store_gc[c]",
      "status" : {
        "state" : "STARTED"
      },
      "description" : "object_store_gc",
      "start_time" : "2026-06-22T13:52:20.627Z",
      "start_time_in_millis" : 1782136340627,
      "running_time" : "1.1m",
      "running_time_in_nanos" : 66709166490,
      "cancellable" : true,
      "cancelled" : false,
      "parent_task_id" : "cluster:6",
      "headers" : {
        "X-Elastic-Project-Id" : "default"
      }
    },{
      "node" : "node_t1",
      "id" : 13,
      "type" : "persistent",
      "action" : "health-node[c]",
      "status" : {
        "state" : "STARTED"
      },
      "description" : "id=health-node",
      "start_time" : "2026-06-22T13:52:20.627Z",
      "start_time_in_millis" : 1782136340627,
      "running_time" : "1.1m",
      "running_time_in_nanos" : 66709619165,
      "cancellable" : true,
      "cancelled" : false,
      "parent_task_id" : "cluster:4",
      "headers" : { }
    },{
      "node" : "node_t1",
      "id" : 14,
      "type" : "persistent",
      "action" : "index-balance-metrics[c]",
      "status" : {
        "state" : "STARTED"
      },
      "description" : "id=index-balance-metrics",
      "start_time" : "2026-06-22T13:52:20.627Z",
      "start_time_in_millis" : 1782136340627,
      "running_time" : "1.1m",
      "running_time_in_nanos" : 66709607063,
      "cancellable" : true,
      "cancelled" : false,
      "parent_task_id" : "cluster:5",
      "headers" : { }
    },{
      "node" : "node_t1",
      "id" : 111,
      "type" : "transport",
      "action" : "internal:index/shard/recovery/stateless_primary_relocation/start",
      "description" : "",
      "start_time" : "2026-06-22T13:52:21.079Z",
      "start_time_in_millis" : 1782136341079,
      "running_time" : "1.1m",
      "running_time_in_nanos" : 66258035127,
      "cancellable" : false,
      "parent_task_id" : "G8rX1mx7RJqAQMRcmgFFew:48",
      "headers" : { }
    }]]
        at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$43(InternalTestCluster.java:2701)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1738)
        at org.elasticsearch.test.InternalTestCluster.assertRequestsFinished(InternalTestCluster.java:2692)
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -584,9 +584,6 @@ tests:
 - class: org.elasticsearch.gradle.internal.InternalDistributionDownloadPluginFuncTest
   method: resolves current version from local build
   issue: https://github.com/elastic/elasticsearch/issues/151851
-- class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
-  method: testCloseWhileShardsAreHollowed
-  issue: https://github.com/elastic/elasticsearch/issues/151861
 - class: org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT
   method: testDeleteIndexAfterRecovery
   issue: https://github.com/elastic/elasticsearch/issues/151884

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessHollowIndexShardsIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessHollowIndexShardsIT.java
@@ -89,6 +89,7 @@ import org.elasticsearch.telemetry.TelemetryProvider;
 import org.elasticsearch.telemetry.TestTelemetryPlugin;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.disruption.NetworkDisruption;
+import org.elasticsearch.test.junit.annotations.TestLogging;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.Transport;
@@ -174,7 +175,6 @@ import static org.elasticsearch.xpack.stateless.recovery.TransportStatelessPrima
 import static org.elasticsearch.xpack.stateless.recovery.TransportStatelessPrimaryRelocationAction.START_RELOCATION_ACTION_NAME;
 import static org.hamcrest.CoreMatchers.either;
 import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.emptyString;
@@ -814,6 +814,14 @@ public class StatelessHollowIndexShardsIT extends AbstractStatelessPluginIntegTe
         assertHitCount(client().prepareSearch(clusterInfo.indexName).setSize(0).setTrackTotalHits(true), clusterInfo.numDocs + moreDocs);
     }
 
+    @TestLogging(value = "org.elasticsearch.xpack.stateless.recovery.TransportStatelessPrimaryRelocationAction:TRACE", reason = """
+        We have seen flakes in this test where assertRequestsFinished() fails during test teardown.
+        See https://github.com/elastic/elasticsearch/issues/151861 for details.
+        In the instance captured there, an internal:index/shard/recovery/stateless_primary_relocation/start task is running during teardown.
+        It seems likely that the race between the index close and the relocation that hollows shards is leaving some asynchronous process
+        hanging so that it never completes its listener chain. This would be a genuine race condition.
+        By enabling trace logging on TransportStatelessPrimaryRelocationAction, we should get more detail about where that task got to.
+        """)
     public void testCloseWhileShardsAreHollowed() throws Exception {
         startMasterOnlyNode();
         final var indexNodeSettings = Settings.builder()

